File: 1009. complement-of-base-10-integer.java
class Solution {
    public int bitwiseComplement(int n) {
        if (n == 0) return 1;
        int mask = (1 << (32 - Integer.numberOfLeadingZeros(n))) - 1;
        return n ^ mask;
    }
    
    public static void main(String[] args) {
        Solution sol = new Solution();
        
        // Example 1:
        int n1 = 5;
        int output1 = sol.bitwiseComplement(n1);
        System.out.println("Example 1: Input: n = " + n1);
        System.out.println("Output: " + output1);
        System.out.println("Expected: 2");
        System.out.println();
        
        // Example 2:
        int n2 = 7;
        int output2 = sol.bitwiseComplement(n2);
        System.out.println("Example 2: Input: n = " + n2);
        System.out.println("Output: " + output2);
        System.out.println("Expected: 0");
        System.out.println();
        
        // Example 3:
        int n3 = 10;
        int output3 = sol.bitwiseComplement(n3);
        System.out.println("Example 3: Input: n = " + n3);
        System.out.println("Output: " + output3);
        System.out.println("Expected: 5");
    }
}

File: 1009. complement-of-base-10-integer.py
class Solution:
    def bitwiseComplement(self, n: int) -> int:
        if n == 0:
            return 1
        bit_length = n.bit_length()
        mask = (1 << bit_length) - 1
        return n ^ mask

if __name__ == "__main__":
    sol = Solution()
    
    # Example 1:
    n = 5
    output = sol.bitwiseComplement(n)
    print(f"Example 1: Input: n = {n}")
    print(f"Output: {output}")
    print(f"Expected: 2")
    print()
    
    # Example 2:
    n = 7
    output = sol.bitwiseComplement(n)
    print(f"Example 2: Input: n = {n}")
    print(f"Output: {output}")
    print(f"Expected: 0")
    print()
    
    # Example 3:
    n = 10
    output = sol.bitwiseComplement(n)
    print(f"Example 3: Input: n = {n}")
    print(f"Output: {output}")
    print(f"Expected: 5")